libs/auth/src/openssl.ts

```diff
@@ -0,0 +1,70 @@
+import { Buffer } from 'buffer';
+import { spawn } from 'child_process';
+import { existsSync } from 'fs';
+import * as fs from 'fs/promises';
+import { v4 as uuid } from 'uuid';
+
+type OpensslParam = string | Buffer;
+
+/**
+ * A convenience function for openssl shell commands. For file params, accepts Buffers containing
+ * the file's contents. Writes these Buffers to temporary files in the specified (or default)
+ * working directory and deletes these files after completion of the openssl command.
+ *
+ * Sample usage:
+ * openssl(['verify', '-CAfile', '/path/to/cert/authority/cert.pem', certToVerifyAsBuffer ]);
+ */
+export async function openssl(
```

Comment:
I took inspiration from https://www.npmjs.com/package/openssl-nodejs when designing this convenience function. I actually originally tried using `openssl-nodejs`, but it had some quirks re where it stored temp files, which params were allowed to be buffers, use of callbacks instead of promises, and type safety. The source code was simple enough that I just decided to create my own function